### DISCUSSION TITLE
Getting selected item from DataGridComboBoxColumn C# XAML

### DISCUSSION THREAD

**Discussion by ivanpbf** [community/community#21380] [Other Feature Feedback, Questions, & Ideas]
<p>Hi I’ve been asking this everywhere was wondering if GitHub Community had the answer</p>
<p>I’ve been trying to get a selected item from a combobox in a datagrid cell.</p>
<p>I have an automatically generated datagrid depending on a list of items. This is my code:</p>
<pre><code>ObservableCollection&lt;Eclass&gt; obsCollection = new ObservableCollection&lt;Eclass&gt;(this.e);
DataGridE.DataContext = obsCollection;
DataGridE.Columns[0].IsReadOnly = true;
DataGridE.Columns[1].Header = "Esp (m)";
DataGridComboBoxColumn descrip = new DataGridComboBoxColumn();
descrip.ItemsSource = new List&lt;string&gt; { "G", "C"};
descrip.Header = "Description";
descrip.TextBinding = new Binding("Description");
DataGridE.Columns[2] = descrip;
//that is the code to generate the grid

//this is the code to get some of the items
 for (int i = 0; i &lt; this.e.Count; i++){
TextBlock esp = DataGridE.Columns[1].GetCellContent(DataGridE.Items[i]) as TextBlock;
</code></pre>
<p>It works great for esp since I can then convert that to a value or a String. But for the combobox column which is the DataGridE.Columns[2] I have no idea how to get the selected item or index, preferably item.</p>
<p>I have tried doing something similar with getcellcontent but it just doesn’t work.</p>
<p>Any help is appreciated.</p>

**Comment by fire-eggs** [comment, marked as answer]
<p>I think <a href="https://social.msdn.microsoft.com/Forums/vstudio/en-US/43d5f87f-1729-4e34-bb8b-e72570dcf419/get-cell-value-of-datagridcombobox-column-for-wpf-datagrid?forum=wpf" rel="nofollow noopener">this</a> has your answer.</p>
<p>To summarize:</p>
<p>GetCellContent returns the contained control. In the case of DataGridComboBoxColumn, the contained control is derived from ComboBox. I.e:</p>
<pre><code>for (int i =0; i &lt; this.e.Count; i++)
{
    ComboBox ele = DataGridE.Columns[2].GetCellContent(DataGridE.Items[i]) as ComboBox;
    var selectedItem = ele.Text;
}
</code></pre>

**Comment by ivanpbf** [comment]
<blockquote>
<blockquote>
<hr>
<blockquote>
<hr>
<p><a class="mention" href="/fire-eggs">@fire-eggs</a> wrote:</p>
<p>I think <a href="https://social.msdn.microsoft.com/Forums/vstudio/en-US/43d5f87f-1729-4e34-bb8b-e72570dcf419/get-cell-value-of-datagridcombobox-column-for-wpf-datagrid?forum=wpf" rel="nofollow noopener">this</a> has your answer.</p>
<p> </p>
<p>To summarize:</p>
<p>GetCellContent returns the contained control. In the case of DataGridComboBoxColumn, the contained control is derived from ComboBox. I.e:</p>
<p> </p>
<p>for (int i =0; i &lt; this.e.Count; i++)<br>
{<br>
ComboBox ele = DataGridE.Columns[2].GetCellContent(DataGridE.Items[i]) as ComboBox;<br>
var selectedItem = ele.Text;<br>
}</p>
<p> </p>
<hr>
</blockquote>
<p>I have asked in different forums, from Stack Overflow to the Visual Studio forum and this is the correct answer to my problem, this worked. Thank you so much!</p>
</blockquote>
</blockquote>
